tools/clang/lib/Sema/SemaExpr.cpp

```diff
@@ -3451,13 +3451,19 @@ ExprResult Sema::ActOnNumericConstant(const Token &Tok, Scope *UDLScope) {
 
       Ty = Context.FloatTy;
     // HLSL Change Starts
-    else if (getLangOpts().HLSL && !Literal.isLong && !Literal.isHalf)
+    else if (getLangOpts().HLSLVersion < hlsl::LangStd::v202x &&
```

Comment:
Why the removal of `getLangOpts().HLSL` conditions around here?  Just the assumption that it's never false for DXC now?  There are some that retain this condition though.

Reply:
I have a preference to keep it, mainly for consistency with the surrounding code and the existing convention.  But I won't die on that hill 😉.  Part of its purpose is to clearly indicate branches that were changed in HLSL-specific ways, and to attempt to limit those change to when this is set.  We did used to have a config where we could build without it enabled, I think, but it's been a long time since any of that would work, I imagine.

It feels weird to partly change to the assumption that it's always true and doesn't need to be checked for just some small subset of the code.  But I don't have a strong reason why you must restore it, since obviously we don't actually support or test any scenario where it's not true.

Also, if comparing the version, HLSL should be known to be true already, otherwise, the result of `getLangOpts().HLSLVersion < hlsl::LangStd::v202x` wouldn't be meaningful, so I don't see how that factors into whether we omit the HLSL check at all.  Perhaps you mean that the version check indicates that this path is HLSL specific to the reader of the code, though I don't think that applies to every omission of the `getLangOpts().HLSL` check.